spec/lib/reporting/identity_verification_report_spec.rb

```diff
@@ -186,6 +242,181 @@
     end
   end
 
+  describe '#fraud_review_passed' do
+    let(:service_provider_for_non_fraud_events) { nil }
+    let(:service_provider_for_fraud_events) { nil }
+
+    let(:cloudwatch_logs) do
+      super().map do |event|
+        is_fraud_event = event['name'].include?('Fraud')
+        event.merge(
+          'service_provider' => is_fraud_event ?
+            service_provider_for_fraud_events :
+            service_provider_for_non_fraud_events,
+        )
+      end
+    end
+
+    context 'when an issuer is specified' do
+      let(:issuer) { 'my:example:issuer' }
+
+      context 'and fraud events are not tagged with sp information' do
+        context 'but other events are tagged for the sp' do
+          let(:service_provider_for_non_fraud_events) { issuer }
+          it 'is users who completed workflow + passed fraud review and any event matches issuer' do
+            expect(report.fraud_review_passed).to eql(3)
+          end
+        end
+
+        context 'and other events are not tagged with an sp' do
+          it 'does not include any users' do
+            expect(report.fraud_review_passed).to eql(0)
+          end
+        end
+
+        context 'and other events are tagged for a different sp' do
+          let(:service_provider_for_non_fraud_events) { 'some:other:sp' }
+
+          it 'does not include any users' do
+            expect(report.fraud_review_passed).to eql(0)
+          end
+        end
+      end
+      context 'and fraud events are tagged with sp information' do
+        let(:service_provider_for_fraud_events) { issuer }
+
+        context 'but other events are not tagged at all' do
```

Comment:
The test (and code) are good, but gosh is this logic complicated. 👼

Reply:
this would probably be a good case for using fixture data pulled from prod and whittled down to only the relevant fields. It's a lot of data but would probably compress pretty well. I'm not sure how everyone would feel about having a .gz file in here

Reply:
[What could go wrong?](https://en.wikipedia.org/wiki/XZ_Utils_backdoor#Mechanism) ;)

Realistically, I think using actual logs could make the tests more, err, _realistic_, but possibly at the expense of being even more complicated to make sense of.